app/src/test/java/ch/hikemate/app/model/facilities/FacilitiesRepositoryTest.kt

```diff
@@ -0,0 +1,334 @@
+import org.junit.Assert.fail
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+import org.mockito.kotlin.any
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.timeout
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+
+class FacilitiesRepositoryTest {
+
+  companion object {
+
+    // Valid JSON Header in JSON Format. Only Valid when the response is closed by
+    // VALID_RESPONSE_HEADER_CLOSER
+    private const val VALID_RESPONSE_HEADER =
+        """
+      {
+        "version": 0.6,
+        "generator": "Overpass API 0.7.62.1 084b4234",
+        "osm3s": {
+          "timestamp_osm_base": "2024-11-21T17:01:26Z",
+          "copyright": "The data included in this document is from www.openstreetmap.org. The data is made available under ODbL."
+        },
+        "elements": ["""
+
+    // Valid JSON Footer in JSON Format. Only Valid when the response is opened by
+    // VALID_RESPONSE_HEADER
+    private const val VALID_RESPONSE_FOOTER = """
+      ]
+      }"""
+
+    // Single, valid amenity element in JSON format
+    private const val VALID_RESPONSE_ELEMENT =
+        """
+      {
+        "type": "node",
+        "id": 3138275148,
+        "lat": 51.0341166,
+        "lon": 7.0464572,
+        "tags": {
+          "amenity": "toilets"
+        }
+      }"""
+  }
+
+  private lateinit var client: OkHttpClient
+  private lateinit var mockCall: Call
+  private lateinit var facilitiesRepository: FacilitiesRepository
+
+  private val callbackCaptor = argumentCaptor<okhttp3.Callback>()
+
+  private val testBoundsNormal = Bounds(46.5, 6.6, 46.51, 6.62)
+
+  private fun buildResponse(request: String): Response {
+    return Response.Builder()
+        .code(200)
+        .message("OK")
+        .body(request.toResponseBody())
+        .protocol(Protocol.HTTP_1_1)
+        .header("Content-Type", "application/json")
+        .request(mock())
+        .build()
+  }
+
+  private fun setupMockResponse(response: Response) {
+    `when`(mockCall.enqueue(callbackCaptor.capture())).then {
+      callbackCaptor.firstValue.onResponse(mockCall, response)
+    }
+  }
+
+  private fun setupNetworkFailure() {
+    `when`(mockCall.enqueue(callbackCaptor.capture())).then {
+      callbackCaptor.firstValue.onFailure(mockCall, IOException("Network failure"))
+    }
+  }
+
+  @Before
+  fun setUp() {
+    client = mock()
+    mockCall = mock()
+
+    facilitiesRepository = FacilitiesRepository(client)
+
+    `when`(client.newCall(any())).thenReturn(mockCall)
+  }
+
+  @Test
+  fun testGetFacilities_integrationTest() =
+      runTest(timeout = 5.seconds) {
+
+        // Calls the real API, not a mocked one
+        val realFacilitiesRepository = FacilitiesRepository(OkHttpClient())
+        realFacilitiesRepository.getFacilities(
+            testBoundsNormal,
+            { routes ->
+              assertNotEquals(routes.size, 0)
+              print(routes)
```

Comment:
Warning: You should not leave print in the code. Either remove it or change it to a log.